backend/src/types.ts

```diff
@@ -339,6 +339,22 @@ export interface ORSection {
   name: string;
 }
 
+/**
+ * A section that has a credit requirement, that can be fulfilled by taking courses in any of the range requirements.
+ * @param type the type of this requirement
+ * @param requirements the possible choices for earning credits
+ * @param numCreditsMin the minimum number of credits needed to satisfy this major
+ * @param numCreditsMax the maximum number of credits needed to satisfy this major
+ * @param name the name of this section
+ */
+export interface RANGESection {
+  type: "RANGE";
+  requirements: Requirement[];
```

Comment:
Since we know that this is a RANGEsection, why not make this just a single `ICourseRange`?